### ISSUE TITLE
bug: Azure Workload Identity doesn't work for Azure SQL Destination due to old go-msssldb reference

### ISSUE THREAD

**Issue by lukerogers** [cloudquery/cloudquery#13269]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When attempting to connect to Azure SQL as a destination using Azure Workload Identity, the following error is shown:

```failed to sync v3 source azure: write client returned error (insert): plugin returned error: DefaultAzureCredential: failed to acquire a token.
Attempted credentials:
	EnvironmentCredential: incomplete environment variable configuration. Only AZURE_TENANT_ID and AZURE_CLIENT_ID are set
	ManagedIdentityCredential: IMDS token request timed out
	AzureCLICredential: Azure CLI not found on path
```
The current go-mssqldb version is v0.19.0 which is using azidentity v1.2.2. The Workload Identity Credential was added to azidentity v1.3.0 which explains why it isn't listed in the attempted credential list above.

The azidentity reference was updated in go-mssqldb v1.1.0.

Note: I know Azure Workload Identity is setup and working as expected since I am using to to read my Azure resources in my source plugin.

### Expected Behavior

Cloudquery should be able to connect to Azure SQL Instances via Azure Workload Identity

### CloudQuery (redacted) config

```yaml
  kind: source
  spec:
    name: "azure"
    registry: "github"
    path: "cloudquery/azure"
    version: "v9.3.1"
    destinations: ["mssql"]
    spec:
      subscriptions: ["****"]
  ---
  kind: destination
  spec:
    name:     "mssql"
    registry: "github"
    path:     "cloudquery/mssql"
    version:  "v4.3.5"
    spec:
      connection_string: "server=SERVER.database.windows.net;port=1433;database=cloudquery;fedauth=ActiveDirectoryDefault"
      auth_mode: "azure"
```

### CloudQuery (redacted) logs

```failed to sync v3 source azure: write client returned error (insert): plugin returned error: DefaultAzureCredential: failed to acquire a token.
Attempted credentials:
	EnvironmentCredential: incomplete environment variable configuration. Only AZURE_TENANT_ID and AZURE_CLIENT_ID are set
	ManagedIdentityCredential: IMDS token request timed out
	AzureCLICredential: Azure CLI not found on path
```

### CloudQuery version

3.1.4

### Pull request (optional)

- [X] I can submit a pull request

**Comment by erezrokah** [comment]
Should be fixed in https://github.com/cloudquery/cloudquery/pull/13272. I'll update once we release a new version with the fix

**Comment by erezrokah** [comment]
This was released in [`v4.3.6`](https://github.com/cloudquery/cloudquery/releases/tag/plugins-destination-mssql-v4.3.6) of the MSSQL destination. If that version doesn't solve the issue please comment and I'll re-open. Thanks for requesting this feature and submitting the PR 🚀
